### ISSUE TITLE
Make contentLabel optional

### ISSUE THREAD

**Issue by bluSCALE4** [reactjs/react-modal#479]
According to the ARIA spec, "If the label text is visible on screen, authors SHOULD use aria-labelledby and SHOULD NOT use aria-label." With that said, since the contentLabel uses aria-label, it shouldn't be required since ideally, if the label is visible on the page, then an id to that element should be provided to aria-labelledby.

https://www.w3.org/TR/wai-aria/states_and_properties#aria-label

**Comment by cycomachead** [comment]
While a more complex validation, I would support something that's either aria-labelledby OR contentLabel is required. I'm migration some existing modals to use the-modal, and this what I'll be implementing on our wrapper component.
